### PR TITLE
Pass `input` as a whole argument to avoid confusion

### PR DESCRIPTION
**Pull request by sayanarijit**
Also, update docs and some minor formatting fixes.

### CODE REVIEW

File: docs/inputs.md [hunk 1]
```diff
@@ -13,11 +13,19 @@ inputs:
...
 
-Input names like ``nixpkgs`` and ``pre-commit-hooks`` are identifiers for what
- is passsed in the first line of the function:
+The dependencies you mention as `inputs` are passsed as an argument to the function.
 
 ```nix title="devenv.nix"
-{ pkgs, lib, nixpkgs, pre-commit-hooks, config, ... }:
+{ inputs, ... }:
+
+{
```
Comment (domenkozar) [inline docs/inputs.md:21]: Let's show an example how to reference for example pre-commit-hooks from the inputs

File: docs/inputs.md [hunk 2]
```diff
@@ -27,6 +35,13 @@ See [basics](basics.md) for more about ``devenv.nix``.
 
 There are a few special inputs passed into ``devenv.nix``:
 
+```nix title="devenv.nix"
+{ pkgs, lib, config, pre-commit-hooks, ... }:
```
Comment (domenkozar) [inline docs/inputs.md:39]: that should be inputs now

File: docs/inputs.md [hunk 3]
```diff
@@ -27,6 +35,13 @@ See [basics](basics.md) for more about ``devenv.nix``.
...
+
+{
+}
+```
+
 - ``pkgs`` is a ``nixpkgs`` input containing all of the available packages for your system.
 - ``lib`` is [a collection of functions for working with Nix data structures](https://nixos.org/manual/nixpkgs/stable/#sec-functions-library).
 - ``config`` is the resolved configuration for your developer environment, which you can use to reference any other options set in ``devenv.nix``.
```
Comment (domenkozar) [inline docs/inputs.md:47]: `inputs.pre-commit-hooks`

File: docs/pre-commit-hooks.md [hunk 4]
```diff
@@ -5,7 +5,7 @@ We recommend a two-step approach for integrating your linters and formatters.
 ## 1) At commit time to make sure that commits are well-formatted
 
 ```nix title="devenv.nix"
-{ pkgs, ... }:
+{ inputs, pkgs, ... }:
```
Comment (domenkozar) [inline docs/pre-commit-hooks.md:8]: Are inputs needed here?

File: docs/inputs.md (hunk 1 above)
Comment (sayanarijit) [reply docs/inputs.md:21]: Getting

```
error: The option `inputs' does not exist. Definition values:
       - In `/home/sayanarijit/Documents/GitHub/devenv/devenv.nix':
           {
             pre-commit = {
               hooks = {
                 nixpkgs-fmt = {
                   enable = true;
           ...

       … while evaluating the attribute 'config'

       at https://github.com/NixOS/nixpkgs/blob/69335c46c48a73f291d5c6f332fb9fe8b8e22b30/lib/modules.nix:330:9:

          329|         options = checked options;
          330|         config = checked (removeAttrs config [ "_module" ]);
             |         ^
          331|         _module = checked (config._module);
```

Comment (sayanarijit) [reply docs/inputs.md:27]: You mean something like https://github.com/cachix/devenv/pull/112/files#diff-cef540503c67fa4a36d1b9fe95a62f979c4b8b2bc43a6e023e250b7154341afcR3-R23?

It's not working btw.

File: flake.nix [hunk 5]
```diff
@@ -14,7 +14,7 @@
   };
   inputs.nix.url = "github:domenkozar/nix/relaxed-flakes";
 
-  outputs = { self, nixpkgs, pre-commit-hooks, nix, ... }:
+  outputs = { self, nixpkgs, pre-commit-hooks, nix, ... }@inputs:
```
Comment (domenkozar) [inline flake.nix:17]:
```suggestion
  outputs = { self, nixpkgs, pre-commit-hooks, nix, ... }:
```
This is not needed :)

File: devenv.nix [hunk 6]
```diff
@@ -1,8 +1,14 @@
-{ pkgs, nix, lib, config, ... }:
+{ inputs, pkgs, lib, config, ... }:
 
+let
```
Comment (domenkozar) [inline devenv.nix:3]: Why do you think we need shellcheck as part of the repo?

File: docs/inputs.md (hunk 1 above)
Comment (domenkozar) [reply docs/inputs.md:27]: That's probably not the best example, maybe we can use an example of using git from a stable version of Nix:

```yaml
inputs:
  nixpkgs-stable:
    url: github:NixOS/nixpkgs/nixpkgs-22.05
```

And then:

```nix
{ inputs, system, ... }:

let 
  nixpkgs-stable = import inputs.nixpkgs-stable { inherit system; };
in {
  packages = [ pkgs-stable.git ];
}
```

**Review by domenkozar — CHANGES_REQUESTED** [review]

File: docs/inputs.md (hunk 1 above)
Comment (sayanarijit) [reply docs/inputs.md:27]: `github:NixOS/nixpkgs/nixpkgs-22.05` doesn't resolve. I think you meant `github:NixOS/nixpkgs/nixos-22.05`. Also, `system` isn't a standalone argument. `pkgs.system` works.

This brings another scope for improvement:

How about we could declare it like:

```yaml
inputs:
  nixpkgs-unstable:  # Defaults to `url: github:NixOS/nixpkgs/nixpkgs-unstable`
  nixos-22.05:  # Defaults to `url: github:NixOS/nixpkgs/nixos-22.05`
```

File: devenv.nix (hunk 6 above)
Comment (sayanarijit) [reply devenv.nix:3]: Thought it was serving as an example. Removing.

File: docs/inputs.md (hunk 1 above)
Comment (domenkozar) [reply docs/inputs.md]: Once we introduce registry this is quite possible

Comment (sayanarijit) [reply docs/inputs.md]: I meant how about letting `url: github:NixOS/nixpkgs` be the default registry? Ultil we have a separate registry field, we have to be specific when adding non github:NixOS/nixpkgs deps.

**Pull request merged**